src/cli/self_update.rs

```diff
@@ -1351,11 +1351,21 @@ fn do_remove_from_path(methods: &[PathUpdateMethod]) -> Result<()> {
 /// rustup-init is stored in `CARGO_HOME`/bin, and then deleted next
 /// time rustup runs.
 pub fn update() -> Result<()> {
-    if NEVER_SELF_UPDATE {
-        err!("self-update is disabled for this build of rustup");
-        err!("you should probably use your system package manager to update rustup");
-        process::exit(1);
```

Comment:
I think we should still check for this - it is the mechanism distros use.

Reply:
I can't believe I lost that check, again good catch, I'll fix that soon.